src/main/java/com/google/cloud/genomics/dataflow/pipelines/ShardedBAMWriting.java

```diff
@@ -115,89 +141,24 @@ public static void main(String[] args) throws GeneralSecurityException, IOExcept
-        int compRefs =  o1.referenceName.compareTo(o2.referenceName);
-        if (compRefs != 0) {
-          return compRefs;
-        }
-        return (int)(o1.start - o2.start);
-      }
-    });
-    final Contig firstContig = contigsList.get(0);
-    
-    // Open and read start of BAM
-    final Storage.Objects storage = Transport.newStorageClient(
-        pipelineOptions
-          .as(GCSOptions.class))
-          .build()
-          .objects();
-    LOG.info("Reading header from " + pipelineOptions.getBAMFilePath());
-    final SamReader samReader = BAMIO
-        .openBAM(storage, pipelineOptions.getBAMFilePath(), ValidationStringency.DEFAULT_STRINGENCY);
-    final SAMFileHeader header = samReader.getFileHeader();
-    
-    LOG.info("Reading first chunk of reads from " + pipelineOptions.getBAMFilePath());
-    final SAMRecordIterator recordIterator = samReader.query(
-        firstContig.referenceName, (int)firstContig.start + 1, (int)firstContig.end + 1, false);
-   
-    Contig firstShard = null;
-    while (recordIterator.hasNext() && result == null) {
-      SAMRecord record = recordIterator.next();
-      final int alignmentStart = record.getAlignmentStart();
-      if (firstShard == null && alignmentStart > firstContig.start && alignmentStart < firstContig.end) {
-        firstShard = shardFromAlignmentStart(firstContig.referenceName, alignmentStart, pipelineOptions.getLociPerWritingShard());
-        LOG.info("Determined first shard to be " + firstShard);
-        result = new HeaderInfo(header, firstShard);
-      }
-    }
-    recordIterator.close();
-    samReader.close();
-    
-    if (result == null) {
-      throw new IOException("Did not find reads for the first contig " + firstContig.toString());
-    }
-    LOG.info("Finished header reading from " + pipelineOptions.getBAMFilePath());
-    return result;
-  }
-
-  /**
-   * Policy used to shard Reads.
-   * By default we are using the default sharding supplied by the policy class.
-   * If you want custom sharding, use the following pattern:
-   * <pre>
-   *    READ_SHARDING_POLICY = new ShardingPolicy() {
-   *     @Override
-   *     public boolean shardBigEnough(BAMShard shard) {
-   *       return shard.sizeInLoci() > 50000000;
-   *     }
-   *   };
-   * </pre>
-   */
-  private static final ShardingPolicy READ_SHARDING_POLICY = ShardingPolicy.BYTE_SIZE_POLICY;
       
   private static PCollection<Read> getReadsFromBAMFile() throws IOException {
-    LOG.info("Sharded reading of "+ pipelineOptions.getBAMFilePath());
+    /**
+     * Policy used to shard Reads.
+     * By default we are using the default sharding supplied by the policy class.
+     * If you want custom sharding, use the following pattern:
+     * <pre>
+     *    BAM_FILE_READ_SHARDING_POLICY = new ShardingPolicy() {
+     *     @Override
+     *     public boolean shardBigEnough(BAMShard shard) {
+     *       return shard.sizeInLoci() > 50000000;
```

Comment:
This is just a code example, the currently used value is already a constant, see ShardingPolicy.MAX_BYTES_PER_SHARD/MAX_LOCI_PER_SHARD constants.
The example is here to show what to do if you want your own decision making about shard sizes.